tools/devtool

```diff
@@ -375,6 +388,102 @@ run_devctr() {
         "$DEVCTR_IMAGE" "${ctr_args[@]}"
 }
 
+# Given a list of packages it adds them to {arch-based}-requirements_freeze.txt
+#
+# Args:
+#   $1  path to the {arch-based}-requirements-freeze file.
+#
+perform_update() {
+    declare requirements_freeze_file_location="$1"
+    shift
+    for package_name in "$@"; do
+        echo "$package_name" >>$requirements_freeze_file_location
+    done
+    echo "" >> $requirements_freeze_file_location
+}
+
+# Looks for any package which are present in requirements.txt file but
+# not in {arch-based}-requirements-freeze.txt file. Prompts the user if they want
+# to install and add freeze their version in {arch-based}-requirements_freeze.txt
+# 
+# Args:
+#   $1  path to the {arch-based}-requirements-freeze file.
+# 
+update_packages() {
+    declare requirements_freeze_file_location="$1"  
+    declare parsed_package_location=$(mktemp)
+    say "Checking for newly added requirements..."
+    while IFS= read -r line; do
+        echo "${line%%=*}" >>$parsed_package_location
+    done <"$requirements_freeze_file_location"
+    unset IFS
+
+    diff=($(grep -v -f $parsed_package_location $FC_DEVCTR_DIR/requirements/requirements.txt))    
+    rm $parsed_package_location
+
+    if [ "${#diff[@]}" -ne 0 ]; then
+        say "List of newly added packages"
+        printf '%s\n' "${diff[@]}"
+        say "Would you like to install and freeze versions of these packages?"
+        get_user_confirmation || return
+        perform_update $requirements_freeze_file_location ${diff[@]}
+    else
+       say "Eveything up-to-date..."
+    fi
+}
+
+# Given an image-id, launches a new docker container.
+# It then runs a version upgrade of outdated packages on container, freezes
+# the new version to a requirements file and copies it back to the host.
+# 
+# Args:
+#   $1  image id 
+#   $2  path to the {arch-based}-requirements-freeze file.
+# 
+perform_upgrade() {
+    declare image_id="$1"
+    declare requirements_freeze_file_location="$2"
+
+    dummy_container_name=$(uuidgen)
+    dummy_container_id=$(docker create -ti --name "$dummy_container_name" "$image_id" bash)    
+    docker start "$dummy_container_id"
+    
+    cmd="python3 -m pip list --outdated --format=freeze | grep -v '^\-e' | cut -d = -f 1 | xargs -n 1 python3 -m pip install -U; python3 -m pip freeze > /tmp/requirements-freeze.txt;"
```

Comment:
please split this very long line into multiple lines. Same story in the `upgrade_version` function

Reply:
fixed across PR.